### PR TITLE
Rough in parallel designspace glyph to IR processing

### PR DESCRIPTION
**Pull request by rsheeter**
Won't *do* anything until #4.

### CODE REVIEW

File: fontc/src/main.rs [hunk 1]
```diff
@@ -52,34 +54,32 @@ impl Config {
     }
 }
 
-fn establish_build_dir() -> PathBuf {
-    let build_dir = Path::new("build");
-    if build_dir.exists() && !build_dir.is_dir() {
-        panic!("The name build is taken by something that isn't a directory");
+fn require_dir(dir: &Path) -> PathBuf {
```
Comment (cmyr) [inline fontc/src/main.rs:57]: this could very reasonably return an error, I'm fine either way

File: fontc/src/main.rs [hunk 2]
```diff
@@ -143,33 +156,93 @@ fn glyphs_deleted<'a>(current_inputs: &Input, prev_inputs: &'a Input) -> HashSet
...
-    let build_dir = establish_build_dir();
-    let (config, prev_inputs) = init(&build_dir, Args::parse()).map_err(Error::IoError)?;
+struct CompileContext {
+    paths: Paths,
+    ir_source: Box<dyn Source>,
+    prev_inputs: Input,
+    current_inputs: Input,
+}
+
+fn establish_context(paths: Paths, args: Args) -> Result<CompileContext, Error> {
```
Comment (cmyr) [inline fontc/src/main.rs:209]: this feels like it would make sense as a type method/constructor?

**Review by cmyr — APPROVED** [review]
My eyes are glazing over a little bit but nothing is jumping out at me 👍

File: fontc/src/main.rs (hunk 2 above)
Comment (rsheeter) [reply fontc/src/main.rs:209]: Hm. I strongly agree! Will fix.

File: fontc/src/main.rs (hunk 1 above)
Comment (rsheeter) [reply fontc/src/main.rs:57]: Hm. In re-reading I agree. Good catch.

**Pull request merged**